src/providers/watsonx.ts

```diff
@@ -319,13 +113,75 @@ export function generateConfigHash(config: any): string {
   return crypto.createHash('md5').update(JSON.stringify(sortedConfig)).digest('hex');
 }
 
-export function calculateWatsonXCost(
+interface ModelSpec {
+  model_id: string;
+  input_tier: string;
+  output_tier: string;
+}
+
+interface WatsonXModel {
+  id: string;
+  cost: {
+    input: number;
+    output: number;
+  };
+}
+
+async function fetchModelSpecs(): Promise<ModelSpec[]> {
+  try {
+    const response = await fetch(
```

Comment:
consider fetchWithCache here to be kinder to your API. You can set a custom expiry key